rkt/prepare.go

```diff
@@ -182,9 +182,20 @@ func runPrepare(cmd *cobra.Command, args []string) (exit int) {
 		Debug:       globalFlags.Debug,
 	}
 
+	fsSupportsOverlay := true
+	if err := common.FSSupportsOverlay(getDataDir()); err != nil {
+		if oerr, ok := err.(common.ErrOverlayUnsupported); ok {
+			stderr.Printf("disabling overlay support: %q", oerr.Error())
+			fsSupportsOverlay = false
+		} else {
+			stderr.PrintE("error determining overlay support", err)
+			return 1
+		}
+	}
+
 	pcfg := stage0.PrepareConfig{
 		CommonConfig:       &cfg,
-		UseOverlay:         !flagNoOverlay && common.SupportsOverlay() && common.FSSupportsOverlay(getDataDir()),
```

Comment:
While being here, `common.SupportsOverlay()` should also return a message, if the `overlay` kernel module is not present. Just stumbled over this.